### PR TITLE
Form views refactor

### PR DESCRIPTION
**Pull request by mannickutd**
Refactor for views, forms and templates trying to use more of the flask functionality.

### CODE REVIEW

File: app/main/forms.py [hunk 1]
```diff
@@ -42,6 +48,16 @@ class RegisterUserForm(Form):
                                          Length(10, 255, message='Password must be at least 10 characters'),
                                          Blacklist(message='That password is blacklisted, too common')])
 
+    def validate_email_address(self, field):
+        # Validate email address is unique.
+        if field.data in self.existing_emails:
+            raise ValidationError('Email address already exists')
+
+    def validate_mobile_number(self, field):
```
Comment (servingUpAces) [inline app/main/forms.py]: We do not have a requirement to keep the mobile number unique.

Comment (mannickutd) [inline app/main/forms.py]: I have commented out the validation check, to be added in later when we need this check.

**Comment by mannickutd** [comment]
Should be good now working with the latest build :).

File: app/main/forms.py [hunk 2]
```diff
@@ -42,6 +48,18 @@ class RegisterUserForm(Form):
...
+        # Validate mobile number is unique
+        # Code to re-added later
+        # if field.data in self.existing_mobiles:
+        #    raise ValidationError('Mobile number already exists')
+        pass
+
 
 class TwoFactorForm(Form):
     sms_code = StringField('sms code', validators=[DataRequired(message='Please enter your code'),
```
Comment (servingUpAces) [inline app/main/forms.py]: Let's use 'Enter verification code' for the label

File: app/main/views/register.py [hunk 3]
```diff
@@ -28,16 +33,16 @@ def process_register():
...
-            return jsonify(database_error='encountered database error'), 400
-    else:
-        return jsonify(form.errors), 400
-    return redirect('/verify')
+        users_dao.insert_user(user)
+        # TODO possibly there should be some exception handling
+        # for sending sms and email codes.
+        # How do we report to the user there is a problem with
+        # sending codes apart from service unavailable?
+        # at the moment i believe http 500 is fine.
```
Comment (servingUpAces) [inline app/main/views/register.py]: Yes we do need more exception handling, we also need to handle exceptions from the db better. There is a story for that.

File: app/main/forms.py (hunk 2 above)
Comment (mannickutd) [inline app/main/forms.py]: Fixed.

**Pull request merged**